### PR TITLE
catch exceptions during database update

### PR DESCRIPTION
**Pull request by spacemansteve**
prevents loss of not yet processed data in rabbitmq message

### CODE REVIEW

**Review by marblestation — CHANGES_REQUESTED** [review]
I think we need to address multiple issues present in this PR:
* We are wrapping a very large portion of code in an `try`
* The exception to be caught is generic and not the concrete one we are targeting (which I assume is just generated at the `commit` step)
* We use the `error` method instead of the `exception` method, which would log the details of the exception
* The error message includes the payload, which can be very large (e.g., fulltext). Bibcode and type might be enough to start future investigations.
* We return a `None` value that will be later used in places such as https://github.com/adsabs/ADSMasterPipeline/blob/master/adsmp/tasks.py#L63, thus logging (at debug level) something like `Saved record from list: None`

I suggest:
* We do the `try` just around the `commit`
* We catch the specific Exception
* We log the problem with `logger.exception` and not include the `payload`
* We control if None is returned in the places we call this update method
* We update the method description to explain that returning None means a problem happened and data was not stored

**Comment by spacemansteve** [comment]
There's a hidden commit that must be covered with a try block, it's in pipeline utils at https://github.com/adsabs/ADSPipelineUtils/blob/master/adsputils/__init__.py#L451.
Since the repercussions of not handling an exception in this code is significant I'd also like to have the json calls covered.  
I agree with the changes to the logger call and updating the documentation.  
I reviewed calls to this code and didn't see where None would cause any problems.  I'm happy to discuss other possible return values.

**Comment by marblestation** [comment]
> https://github.com/adsabs/ADSPipelineUtils/blob/master/adsputils/__init__.py#L451

After the explicit commit within the `with` statement, if there is an exception, you could rollback so the second implicit commit from the `with` statement would not have any effect.

If by capturing JSON exception you mean the line where we call `out = r.toJSON()`, that's right next to the commit and also a different non-generic Exception could be used. Still, no need to wrap everything in a large `try`.

As I mentioned, the problem with returning `None` is producing a log message at debug level saying: `Saved record from list: None` (e.g., https://github.com/adsabs/ADSMasterPipeline/blob/master/adsmp/tasks.py#L63,), when the message could be more clear.

I see value in implementing these rather small changes to improve readability and maintenance.

File: adsmp/app.py [hunk 1]
```diff
@@ -150,12 +152,17 @@ def update_storage(self, bibcode, type, payload):
...
-            return out
-
+            try:
+                session.commit()
+                return out
+            except exc.IntegrityError:
+                msg = 'error in app.update_storage while updating database for bibcode {}, type {}'.format(bibcode, type)
+                self.logger.exception(msg)
+                session.rollback()
+                return msg
```
Comment (marblestation) [inline adsmp/app.py:165]: I did not have an issue returning `None` but about what was done with that value in places like https://github.com/adsabs/ADSMasterPipeline/blob/master/adsmp/tasks.py#L63. Returning a string with success and failure makes it harder to know if the method succeeded from outside.

**Review by marblestation — CHANGES_REQUESTED** [review]
I am sorry @spacemansteve but returning `None` on failures made it easier to detect when the method has failed, while now we always return strings and it is more tricky to know if the storage was updated or not. My concern was about the `logger.debug` calls after the update storage method is called and not about returning `None` per se.

**Review by marblestation — APPROVED** [review]
Thanks!

**Pull request merged**